### DISCUSSION TITLE
Support for new assistant api

### DISCUSSION THREAD

**Discussion by eburnette** [danny-avila/LibreChat#1323] [Feature Requests & Suggestions]
I see some code to use the new OpenAI Assistant API (Threads, runs, etc.) but so far I haven't found a way to use it in the client, can you point me in the right direction?

**Comment by danny-avila** [comment]
> I see some code to use the new OpenAI Assistant API (Threads, runs, etc.) but so far I haven't found a way to use it in the client, can you point me in the right direction?

It should not be accessible at the moment. It's a work in progress and I'm actively developing it.

**Reply by taymossimc** [reply]
Just curious where this project stands at the moment? How's it going? I would love to see the OpenAI Assistant API supported in LibreChat.  Thanks!

**Reply by danny-avila** [reply]
Nearing completion, hope to be done tomorrow!

![image](https://github.com/danny-avila/LibreChat/assets/110412045/cd085cb9-41d9-48bc-9d5c-4103e4ba43f4)

**Reply by taymossimc** [reply]
Nice! I'm pleased to hear that! Thanks for you hard work on this project.

**Comment by louis030195** [comment]
https://github.com/stellar-amenities/assistants @eburnette

**Reply by danny-avila** [reply]
@louis030195 Interesting reminds me of [opengpts](https://github.com/langchain-ai/opengpts), and [openopenai](https://github.com/transitive-bullshit/OpenOpenAI). 

What makes this library different (other than using non-OpenAI models), which local llms/apis are supported, and could i theoretically use OpenAI models (no reverse proxy) for some benefit with this library?

**Reply by louis030195** [reply]
1. difference with opengpts and openopenai is the energy x direction. `stellar-amenities/assistants` is intended to be used in spacecrafts in space in q4 2024. I don't know if langchain will maintain their nor the other repo
2. atm all (local) llms that have openai like chat completion + openai + anthropic
3. yes you can use openai. advantage would be more data control, different performance (working on benchmarks), more customisation (while staying identical to openai experience)


OpenAI has the best brains in the world, copying them but in a different way is just cost efficient. Vicarious learning

**Reply by danny-avila** [reply]
>  stellar-amenities/assistants is intended to be used in spacecrafts in space in q4 2024. I don't know if langchain will maintain their nor the other repo

can you tell me more about this? And a source on this if there’s an article or something?

> OpenAI has the best brains in the world, copying them but in a different way is just cost efficient. Vicarious learning

This has been a driving force for LibreChat since the beginning!
